### PR TITLE
Add option to control how custom field options are ordered.

### PR DESCRIPTION
**Pull request by balsdorf**
Historically custom field options have been ordered alphabetically. I've now added the option to order them:
* Alphabetically / Reverse Alphabetically
* Insert / Reverse Insert
* Manually

Custom field options have also been moved to a separate page to allow for ordering and editing.

### CODE REVIEW

File: src/Controller/Manage/CustomFieldOptionsController.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    protected function configure()
+    {
+        $request = $this->getRequest();
+
+        $this->fld_id = $request->request->get('fld_id') ?: $request->query->get('fld_id');
+        $this->field_info = Custom_Field::getDetails($this->fld_id);
+
+        $this->cat = $request->request->get('cat') ?: $request->query->get('cat');
+
+        if (empty($this->field_info)) {
```
Comment (glensc) [inline src/Controller/Manage/CustomFieldOptionsController.php]: move this to "action" method

File: upgrade/patches/64_custom_field_option_order.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Control the order of custom fields
+ *
+ */
+ALTER TABLE {{%custom_field_option}} ADD COLUMN cfo_rank int(10) NOT NULL DEFAULT 0;
```
Comment (glensc) [inline upgrade/patches/64_custom_field_option_order.sql]: avoid using WIDTH to integer columns, use just `INT`, as it's misleading, also, if you don't need negative values, use `UNSIGNED INT`. and perhaps even `UNSIGNED TINYINT` if 255 values is enough (or `SMALLINT`)

also use `AFTER` or `BEFORE` to order numeric fields first in the table.

Comment (balsdorf) [inline upgrade/patches/64_custom_field_option_order.sql]: Old habits die hard. Fixed.

File: src/Controller/Manage/CustomFieldOptionsController.php (hunk 1 above)
Comment (balsdorf) [inline src/Controller/Manage/CustomFieldOptionsController.php]: I've moved this to prepareTemplate as I need it on the view page, not the update action.

File: upgrade/patches/64_custom_field_option_order.sql [hunk 3]
```diff
@@ -2,5 +2,5 @@
  * Control the order of custom fields
  *
  */
-ALTER TABLE {{%custom_field_option}} ADD COLUMN cfo_rank int(10) NOT NULL DEFAULT 0;
+ALTER TABLE {{%custom_field_option}} ADD COLUMN cfo_rank INT UNSIGNED NOT NULL DEFAULT 0 AFTER cfo_fld_id;
 ALTER TABLE {{%custom_field}} ADD COLUMN fld_order_by varchar(20) NOT NULL DEFAULT 'cfo_id ASC'
```
Comment (glensc) [inline upgrade/patches/64_custom_field_option_order.sql]: always add some `AFTER`, `BEFORE` keyword

also, please avoid using useless commit messages like `Clean up code from code review`, such message does not explain purpose or changes of the commit. in other words it's useless later if looking at git logs what changed.

File: src/Controller/Manage/CustomFieldOptionsController.php [hunk 4]
```diff
@@ -42,13 +42,8 @@ protected function configure()
         $request = $this->getRequest();
 
         $this->fld_id = $request->request->get('fld_id') ?: $request->query->get('fld_id');
-        $this->field_info = Custom_Field::getDetails($this->fld_id);
```
Comment (glensc) [inline src/Controller/Manage/CustomFieldOptionsController.php]: the class property can be removed now as well

**Comment by balsdorf** [comment]
Anything else you would like seen done on this or can I merge this in?

**Comment by glensc** [comment]
missing `AFTER`, `BEFORE` keywords in sql. you fixed only one query.

**Comment by balsdorf** [comment]
I want that column at the end of the table so an after clause is superfluous. None of the patches after 17 use it either.

I'm fine with using BEFORE / AFTER to group ints at the start of the table or to group related fields together but in this case it is totally pointless.

**Comment by glensc** [comment]
uh, this PR messed up (reset to default?) existing fields in `list.php` view?

image

**Comment by balsdorf** [comment]
This PR did not touch that area. I can't reproduce the problem.

**Pull request merged**